browser/ephemeral_storage/ephemeral_storage_tab_helper.cc

```diff
@@ -0,0 +1,139 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ephemeral_storage/ephemeral_storage_tab_helper.h"
+
+#include <map>
+#include <set>
+
+#include "base/feature_list.h"
+#include "base/hash/md5.h"
+#include "base/no_destructor.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/session_storage_namespace.h"
+#include "content/public/browser/storage_partition.h"
+#include "content/public/browser/web_contents.h"
+#include "net/base/registry_controlled_domains/registry_controlled_domain.h"
+#include "third_party/blink/public/common/features.h"
+
+using content::BrowserContext;
+using content::NavigationHandle;
+using content::SessionStorageNamespace;
+using content::WebContents;
+
+namespace ephemeral_storage {
+
+namespace {
+
+using PerTLDEphemeralStorageMap =
+    std::map<PerTLDEphemeralStorageKey, base::WeakPtr<PerTLDEphemeralStorage>>;
+
+PerTLDEphemeralStorageMap& active_per_tld_storage_areas() {
+  static base::NoDestructor<PerTLDEphemeralStorageMap> active_storage_areas;
+  return *active_storage_areas.get();
+}
+
+std::string URLToStorageDomain(const GURL& url) {
+  std::string domain = net::registry_controlled_domains::GetDomainAndRegistry(
+      url, net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
+
+  // GetDomainAndRegistry might return an empty string if this host is an IP
+  // address or a file URL.
+  if (domain.empty())
+    domain = url::Origin::Create(url.GetOrigin()).Serialize();
+
+  return domain;
+}
+
+// Session storage ids are expected to be 36 character long GUID strings. Since
+// we are constructing our own ids, we convert our string into a 32 character
+// hash and then use that make up our own GUID-like string. Because of the way
+// we are constructing the string we should never collide with a real GUID and
+// we only need to worry about hash collisions, which are unlikely.
+std::string StringToSessionStorageId(const std::string& string) {
```

Comment:
this function can append `"/ephemeral-local-storage"` to the string

Reply:
One caller needs the string "/ephemeral-local-storage" and the other needs "/ephemeral-session-storage". What do you think about adding an additional parameter which is the suffix to be added to the string?

Reply:
I've gone ahead and done this in both `dom_window_storage.cc` and `ephemeral_storage_tab_helper.cc`.